test/controllers/email/VerifyPasscodeControllerSpec.scala

```diff
@@ -538,17 +562,29 @@ class VerifyPasscodeControllerSpec extends ControllerBaseSpec with MockEmailVeri
       "the emailPinVerification feature switch is enabled" when {
 
         "there is an email in session" should {
+          lazy val updateEmailMockResponse = Future.successful(Right(UpdateEmailSuccess("success")))
```

Comment:
Can you also add tests for the `updateContactPrefEmail` action when calling the service functions to check email verification and update the email. Currently untested scenarios are: email not yet verified, conflict response, unexpected error response. Looks like some of these scenarios are split out into a separate function, `sendUpdateRequest`, so it's up to you if you want to split out the tests or just keep them all in this action.